File: 1752-check-if-array-is-sorted-and-rotated/solution.cpp
#include <vector>

class Solution {
public:
    bool check(std::vector<int>& nums) {
        int count = 0;
        int n = nums.size();
        
        for (int i = 0; i < n; ++i) {
            // Compare current element with the next element in a circular manner
            if (nums[i] > nums[(i + 1) % n]) {
                count++;
            }
            // If we have more than one rotation boundary (drop point), it's invalid
            if (count > 1) {
                return false;
            }
        }
        
        return true;
    }
};

File: 1752-check-if-array-is-sorted-and-rotated/solution.py
class Solution:
    def check(self, nums: list[int]) -> bool:
        count = 0
        n = len(nums)
        
        for i in range(n):
            # Compare current element with the next element (circular array index)
            if nums[i] > nums[(i + 1) % n]:
                count += 1
                
            # If we find more than one inversion, the array cannot be sorted and rotated
            if count > 1:
                return False
                
        return True